content/docs/guides/custom-name-resolution.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Custom Name Resolution
+description: >-
+  Explains standard name resolution, the custom name resolver interface,
+  and how to write an implementation.
+---
+
+### Overview
+
+Name resolution is fundamentally about service discovery. When sending a request
+to `grpc.io`, the client must determine the IP address of that service name.
+Name resolution is often thought to be the same as
+[DNS](https://www.ietf.org/rfc/rfc1035.txt). In practice however, DNS is usually
+augmented with extensions or completely replaced to enable name resolution.  For
+example, the `/etc/hosts` file on Posix systems is not strictly speaking part of
+DNS and instead stems from
+[RFC-952](https://datatracker.ietf.org/doc/html/rfc952).
+
+When making a request with a gRPC client, by default, DNS name resolution is
+used. However, various other name resolution mechanisms may be used:
+
+|Resolver|Example|Notes|
+|-|-|-|
+|DNS|`grpc.io:50051`|By default, DNS is assumed.|
+|DNS|`dns:///grpc.io:50051`|The extra slash is used to provide an authority|
+|Unix Domain Socket|`uds:///var/run/docker.sock`|
```

Comment:
Do we really want to use `docker` here? It isn't using grpc.

Reply:
I know we discussed this offline and I said that the docker socket now uses gRPC. This may not be right -- it's the containerd socket which speaks gRPC. I've updated this example accordingly.